tree/ntuple/src/RNTupleDescriptorFmt.cxx

```diff
@@ -124,7 +124,14 @@ void ROOT::RNTupleDescriptor::PrintInfo(std::ostream &output) const
          }
          const auto &pageRange = cluster.second.GetPageRange(column.second.GetPhysicalId());
          auto idx = cluster2Idx[cluster.first];
+         std::uint64_t prevLocatorOffset = 0;
+         std::uint64_t locatorOffset;
          for (const auto &page : pageRange.GetPageInfos()) {
+            locatorOffset = page.GetLocator().GetType() == ROOT::RNTupleLocator::ELocatorType::kTypeDAOS
+                               ? page.GetLocator().GetPosition<RNTupleLocatorObject64>().GetLocation()
+                               : page.GetLocator().GetPosition<std::uint64_t>();
+            if (locatorOffset == prevLocatorOffset)
```

Comment:
I'm pretty sure that `pageRange.GetPageInfos()` is not guaranteed to return the pages in sorted order, so you might miss some duplicates here if they are not adjacent. Usually they will be but it's not a guarantee (and I observed this is the rntviewer)

Reply:
I think it has, we are also deduplicating pages between page ranges of different columns. This might even be the primary case, for example for two index columns of two collections that always have the same number of elements.